src/imageops/transform.rs

```diff
@@ -0,0 +1,65 @@
+extern crate rayon;
+use self::rayon::prelude::*;
+
+use decoders::{Orientation, RawImage};
+use imageops::OpBuffer;
+
+/// Mirror an OpBuffer horizontally
+pub fn flip_horizontal(buf: &OpBuffer) -> OpBuffer {
+  let mut out = OpBuffer::new(buf.width, buf.height, buf.colors);
+  out.data.par_chunks_mut(out.width * out.colors).enumerate().for_each(|(row, line)| {
+    let offset = buf.width * row * buf.colors;
+    for col in 0 .. buf.width {
+      for c in 0 .. buf.colors {
+        line[col * buf.colors + c] = buf.data[offset + (buf.width - 1 - col) * buf.colors + c];
+      }
+    }
+  });
+
+  out
+}
+
+/// Mirror an OpBuffer vertically
+pub fn flip_vertical(buf: &OpBuffer) -> OpBuffer {
+  let mut out = OpBuffer::new(buf.width, buf.height, buf.colors);
+  out.data.par_chunks_mut(out.width * out.colors).enumerate().for_each(|(row, line)| {
+    let offset = (buf.height - 1 - row) * buf.width * buf.colors;
+    for col in 0 .. buf.width * buf.colors {
+      line[col] = buf.data[offset + col];
+    }
+  });
+
+  out
+}
+
+/// Transpose an OpBuffer
+pub fn transpose(buf: &OpBuffer) -> OpBuffer {
+  let mut out = OpBuffer::new(buf.height, buf.width, buf.colors);
+
+  out.data.par_chunks_mut(out.width * out.colors).enumerate().for_each(|(row, line)| {
+    for col in 0 .. buf.height {
+      let target = col * buf.colors;
+      let source = (col * buf.width + row) * buf.colors;
+      for c in 0 .. buf.colors {
+        line[target + c] = buf.data[source + c];
+      }
+    }
+  });
+
+  out
+}
+
+/// Rotate an OpBuffer based on the given RawImage's orientation
+pub fn rotate(img: &RawImage, buf: &OpBuffer) -> OpBuffer {
+  match img.orientation {
+    Orientation::HorizontalFlip => flip_horizontal(buf),
+    Orientation::Rotate180 => flip_horizontal(&flip_vertical(buf)),
```

Comment:
I had a bit of time to think about this and it doesn't seem too hard to do it in one go. Here's some example code that should work (not even compile tested yet):

https://gist.github.com/pedrocr/4e65061da3dfa1af5b2c2fe2ac5e149e

This kind of code is tricky enough that it makes sense to have some unit tests for it. It should be simple enough to have some 3x4 pixel OpBuffer test examples for each of the orientation cases.